src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -144,6 +151,60 @@ export const txHistory = createSlice({
       state.status = payload
     },
     onMessage: (txState, { payload }: TxMessage) =>
-      updateOrInsert(txState, payload.message, payload.accountSpecifier)
+      updateOrInsert(txState, payload.message, payload.accountSpecifier),
+    upsertTxs: (txState, { payload }: TxsMessage) => {
+      for (const tx of payload.txs) {
+        updateOrInsert(txState, tx, payload.accountSpecifier)
```

Comment:
it would be nice to write a new reducer to do a single state update with all the new txids, but this will work for now and we can address performance later